src/components/CardGrid.tsx

```diff
@@ -25,32 +26,67 @@ export interface CardGridWithHeaderProps<T> {
   search: SearchBarProps;
 
   menus?: JSX.Element | JSX.Element[];
+
+  filters?: FilterProps<F>[];
 }
 
 /**
  * Grid layout component with triggers for fetching more data for infinite scroll, search bar, and grid size selection.
  *
  * @template T type of data being fetched and used to create cards
+ * @template F type for filter options
  * @param props
  * @returns
  */
-export function CardGridWithHeader<T>(props: CardGridWithHeaderProps<T>): JSX.Element {
-  const [gridView, setGridView] = useState<GridView>(DEFAULT_GRID_VIEW);
+export function CardGridWithHeader<T, F = null>(props: CardGridWithHeaderProps<T, F>): JSX.Element {
+  const [collapsed, setCollapsed] = useState<boolean>(false);
+  const gridViews = useMemo(() => {
+    return ifElse(
+      isNil,
+      always([GridView.FOUR_BY_FOUR, GridView.FIVE_BY_FIVE]),
+      ifElse(
+        always(collapsed),
+        always([GridView.FOUR_BY_FOUR, GridView.FIVE_BY_FIVE]),
+        always([GridView.THREE_BY_THREE, GridView.FOUR_BY_FOUR])
+      )
+    )(props.filters);
+  }, [props.filters, collapsed]);
```

Comment:
@kristianeboe I know I promised to not used ramda but its why to productive to not leverage. I'd like to do a workshop or 2 on the topic to kick start folks familiarity with the package.

Reply:
I'm sorry, but I want to push back on that. Even if you could bring our team up to speed on Ramda, you can't hold a workshop for individual contributors who want to engage with and contribute to the open source code base. 

Personally I dont't easily understand what's going on here either, but maybe it's just late in the evening for me.

As for the grid views they might need some rethinking in its entirety, a potentially better solution is to define tailwind grid classes based on type of grid so:
base: grid gap-6
profileGrid: grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4
nftGrid: grid-cols-2 sm:grid-cols-4 md:grid-cols-6 lg:grid-cols-8
etc.

We can discuss more ofc.

Reply:
I removed the use of ramda. And we should keep our grid options consistent poor.